app/src/main/java/org/wikipedia/page/PageProperties.java

```diff
@@ -27,34 +30,60 @@
     private final long revisionId;
     private final Date lastModified;
     private final String displayTitleText;
-    private final String editProtectionStatus;
+    private String editProtectionStatus;
     private final int languageCount;
     private final boolean isMainPage;
     private final boolean isDisambiguationPage;
     /** Nullable URL with no scheme. For example, foo.bar.com/ instead of http://foo.bar.com/. */
-    @Nullable private final String leadImageUrl;
-    @Nullable private final String leadImageName;
-    @Nullable private final String titlePronunciationUrl;
+    @Nullable private String leadImageUrl;
+    @Nullable private String leadImageName;
     @Nullable private final Location geo;
     @Nullable private final String wikiBaseItem;
     @Nullable private final String descriptionSource;
+    @Nullable private Protection protection;
 
     /**
      * True if the user who first requested this page can edit this page
      * FIXME: This is not a true page property, since it depends on current user.
      */
-    private final boolean canEdit;
+    private boolean canEdit;
 
     /**
      * Side note: Should later be moved out of this class but I like the similarities with
      * PageProperties(JSONObject).
      */
+    public PageProperties(@NonNull PageSummary pageSummary) {
+        pageId = pageSummary.getPageId();
+        namespace = pageSummary.getNamespace();
+        revisionId = pageSummary.getRevision();
+        displayTitleText = defaultString(pageSummary.getDisplayTitle());
+        geo = pageSummary.getGeo();
+        languageCount = 1;
```

Comment:
@sharvaniharan  Were you going to update this to receive the language count from somewhere else?
(cc @joewalsh )